[#67] Configurations for Setting Compaction Strategies for Any Tables

jparkie opened the pull request:
https://github.com/krasserm/akka-persistence-cassandra/issues/67

I am creating this pull-request to ask for a code review for my proposed solution. I also need advice on the depth of tests I should write for this solution.

~~I haven't had the actual chance to try out my solution due to other commitments. Thus, please do not merge this request yet.~~

Edit: I have provided some basic test coverage, and I have cleaned my commit messages.

krasserm commented:
@jparkie thanks a lot for your contribution, looks good to me at first glance. I wonder if we really need compaction for the metadata and config table as we don't store much data in it. Having it for the journal table should be sufficient IMO. WDYT?

jparkie commented:
I also agree. I feel compaction settings for the journal to be enough as it will be the most write heavy.

jparkie commented:
@krasserm I have simplified the settings for the CompactionStrategy to allow only journal and snapshot tables. I have written some tests. I have also squashed the commits from before. I am happy if this gets merged. :)

krasserm commented:
@jparkie great work - thanks a lot. Would you mind adding the configuration options to the documentation in README (including a hint that it requires Cassandra 2.2, does it)? I'll merge afterwards.

jparkie commented:
It doesn't require Cassandra 2.2, it is backwards compatible. :)

krasserm commented:
Fantastic :smiley:

jparkie commented:
@krasserm I have updated the README.md, I referenced both the CQL documentation and the tests as resources and examples to configure the CompactionStrategy. :)

krasserm commented:
@jparkie thanks a lot for the contribution and keep up the good work! :smiley:
